### DISCUSSION TITLE
"if: github.ref == x" while not on the branch now causes a build failure

### DISCUSSION THREAD

**Discussion by cfarrend** [community/community#25128] [Actions]
<p>Hi,</p>
<p>Wondering if anyone is experiencing the same issue we have?</p>
<p>We have a workflow defined as follows:</p>
<pre><code>name: Push workflow

on: push

jobs:
  master_task:
    if: github.ref == 'refs/heads/master'
    runs-on: ubuntu-latest
    steps:
      ...
  branch_task:
    runs-on: ubuntu-latest
    steps:
      ...
</code></pre>
<p>Previously, when this was run on a branch that is not master, the master_task job would not run and skip (<a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif</a>) while the branch_task would run</p>
<p>Now, as of around 24 hours ago, most of our repositories seem to be showing the check as a fail instead of skipping with no useful output. The most strange thing is that in a PR the check will show as a failure, however in the actions console it will show as a success.</p>
<p>(example from our own repo)</p>
<p><strong>Pull Request</strong></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/0bd24a2ff50c6ef222ce4240238b67154739909e.png" data-download-href="/uploads/short-url/1GzNlKCM531os8UHG9ue2QMmB78.png?dl=1" title="Screen Shot 2020-04-07 at 11.00.17 am.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12799795/181089049-37102bb3-86d1-4bfb-81c8-fe88ea287838.png" alt="Screen Shot 2020-04-07 at 11.00.17 am.png" data-base62-sha1="1GzNlKCM531os8UHG9ue2QMmB78" width="690" height="36" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/0/0bd24a2ff50c6ef222ce4240238b67154739909e_2_690x36.png, https://aws1.discourse-cdn.com/github/optimized/2X/0/0bd24a2ff50c6ef222ce4240238b67154739909e_2_1035x54.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/0/0bd24a2ff50c6ef222ce4240238b67154739909e_2_1380x72.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/0bd24a2ff50c6ef222ce4240238b67154739909e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-04-07 at 11.00.17 am.png</span><span class="informations">1924×102 12.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/206d66c3b74dcb9db21f876e09217d7a5ac5bbec.png" data-download-href="/uploads/short-url/4CRFzuEG6pbTP5k0S2gG8txhR92.png?dl=1" title="Screen Shot 2020-04-07 at 11.00.44 am.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12799795/181089051-f914e462-b21d-4f4f-8c96-539287f2cf34.png" alt="Screen Shot 2020-04-07 at 11.00.44 am.png" data-base62-sha1="4CRFzuEG6pbTP5k0S2gG8txhR92" width="524" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/206d66c3b74dcb9db21f876e09217d7a5ac5bbec_2_524x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/2/206d66c3b74dcb9db21f876e09217d7a5ac5bbec_2_786x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/2/206d66c3b74dcb9db21f876e09217d7a5ac5bbec_2_1048x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/206d66c3b74dcb9db21f876e09217d7a5ac5bbec_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-04-07 at 11.00.44 am.png</span><span class="informations">1484×1414 255 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Note that the upload_manifests task has failed</p>
<p><strong>GitHub Actions view</strong></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/f/f1ef5ee81618e6a2bc5a2602f0183905001321f3.png" data-download-href="/uploads/short-url/ywfV1HZlHsYpSMDhenvDVJunKZJ.png?dl=1" title="Screen Shot 2020-04-07 at 10.59.04 am.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12799795/181089052-96eadbb1-4fcc-435e-9af1-8c5a4dffc5d9.png" alt="Screen Shot 2020-04-07 at 10.59.04 am.png" data-base62-sha1="ywfV1HZlHsYpSMDhenvDVJunKZJ" width="690" height="52" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/f/f1ef5ee81618e6a2bc5a2602f0183905001321f3_2_690x52.png, https://aws1.discourse-cdn.com/github/optimized/2X/f/f1ef5ee81618e6a2bc5a2602f0183905001321f3_2_1035x78.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/f/f1ef5ee81618e6a2bc5a2602f0183905001321f3_2_1380x104.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/f/f1ef5ee81618e6a2bc5a2602f0183905001321f3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-04-07 at 10.59.04 am.png</span><span class="informations">1852×140 18.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div> <div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/5/5484d69d645a01252afcefe3988076b7d93d6b4e.png" data-download-href="/uploads/short-url/c3GIag9W5972UgeI6skmw57V6Ka.png?dl=1" title="Screen Shot 2020-04-07 at 10.59.16 am.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12799795/181089054-a5de06eb-ffa4-47cc-8c4c-7b4fabb58364.png" alt="Screen Shot 2020-04-07 at 10.59.16 am.png" data-base62-sha1="c3GIag9W5972UgeI6skmw57V6Ka" width="690" height="254" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/5/5484d69d645a01252afcefe3988076b7d93d6b4e_2_690x254.png, https://aws1.discourse-cdn.com/github/optimized/2X/5/5484d69d645a01252afcefe3988076b7d93d6b4e_2_1035x381.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/5/5484d69d645a01252afcefe3988076b7d93d6b4e_2_1380x508.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/5484d69d645a01252afcefe3988076b7d93d6b4e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-04-07 at 10.59.16 am.png</span><span class="informations">2858×1056 113 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Note in this case, the upload_manifests task has failed with no output while it should have skipped. This was happening a few days ago and nothing was changed to the actions file.</p>
<p>Just wondering if there is something our end that may have potentially changed or something up with GitHub Actions?</p>
<p>Thanks,</p>
<p>Chris</p>

**Comment by Yanjingzhu** [comment]
<p>Hi <a class="mention" href="/cfarrend">@cfarrend</a> , </p>
<p>I tested in my side, skipped job is not showing in the workflow run page, and the workflow does not failed. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/060abd1084f387c80b5f2514445f12388948bf12.png" data-download-href="/uploads/short-url/RrS43MMG6xcqFSMLoAZR0lPdPc.png?dl=1" title="push.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181089057-29a579d6-3741-48f2-989e-0832c2cd15ee.png" alt="push.png" data-base62-sha1="RrS43MMG6xcqFSMLoAZR0lPdPc" width="690" height="221" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/0/060abd1084f387c80b5f2514445f12388948bf12_2_690x221.png, https://aws1.discourse-cdn.com/github/optimized/2X/0/060abd1084f387c80b5f2514445f12388948bf12_2_1035x331.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/0/060abd1084f387c80b5f2514445f12388948bf12.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/060abd1084f387c80b5f2514445f12388948bf12_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">push.png</span><span class="informations">1180×378 17.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>There is the branch policy of master branch: </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/74c5f804abe40652a09c8d9fbf50f9c73614bbaa.png" data-download-href="/uploads/short-url/gF1x28RtOwXEqPQXziWf4o6Ta5A.png?dl=1" title="branch policy.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181089058-f0b549fc-84cf-400b-9efb-c90b4c962954.png" alt="branch policy.png" data-base62-sha1="gF1x28RtOwXEqPQXziWf4o6Ta5A" width="577" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/74c5f804abe40652a09c8d9fbf50f9c73614bbaa_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">branch policy.png</span><span class="informations">751×650 35.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>And I merged my PR with admin privilege. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/5/5cdb3c3726d009b4675b506ca92d2549eb9d77b4.png" data-download-href="/uploads/short-url/dfrDsXCjNUP16MXTlqT9DME6t6Y.png?dl=1" title="merge admin.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181089060-094bc2d0-c72a-4fd4-9cc4-f40d6e20a101.png" alt="merge admin.png" data-base62-sha1="dfrDsXCjNUP16MXTlqT9DME6t6Y" width="690" height="411" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/5cdb3c3726d009b4675b506ca92d2549eb9d77b4_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">merge admin.png</span><span class="informations">765×456 22.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Could you please check the current behavior in your side?  It would be better if you could share me your branch policy settings. </p>

**Comment by cfarrend** [comment]
<p><a class="mention" href="/yanjingzhu">@yanjingzhu</a> yes this is what we normally expect, and it has been the case for months now. Issue is that recently we are now getting errors for this so we are wondering if anything has changed for GitHub actions in relation to our account.</p>
<p>I only mention this as we have been a (unfortunate) victim of actions related issues before:</p>
<p><a href="https://github.community/t/github-actions-not-running-on-our-organisation-repositories/17624/9">https://github.community/t5/GitHub-Actions/GitHub-Actions-not-running-on-our-organisation-repositories/m-p/48375</a></p>
<p>In regards to branch policy settings, we have no rules on it</p>

**Comment by Yanjingzhu** [comment]
<p><a class="mention" href="/cfarrend">@cfarrend</a>  I could not reproduce your issue in my side. If this issue happened in a private repo,I would recommend you to submit a support ticket to contact <a href="https://support.github.com/contact" rel="nofollow noopener">GitHub Support</a> directly. </p>

**Comment by cfarrend** [comment]
<p>I am afraid that would be the case <a class="mention" href="/yanjingzhu">@yanjingzhu</a>, issue is that GitHub support tickets in the past take usually 20 days for a response so was wondering if it could somehow be expedited via GitHub actions forums as it’s a bit of an annoyance to our development process, similar to the last ticket</p>

**Comment by Yanjingzhu** [comment]
<p>I am sorry to hear that . Could you please share your organization name and repo name here? </p>

**Comment by susannemoog** [comment]
<p>We have exactly the same problem and I have an example in a public repository: <a href="https://github.com/TYPO3-Documentation/sphinx_typo3_theme/pull/69" rel="nofollow noopener">https://github.com/TYPO3-Documentation/sphinx_typo3_theme/pull/69</a> - if anyone wants to take a look, I’d be grateful.</p>

**Comment by AlekSi** [comment]
<p>Same problem here with all our private repositories. Hopefully, that change in behavior will be reverted.</p>

**Comment by cfarrend** [comment]
<p>Hey <a class="mention" href="/yanjingzhu">@yanjingzhu</a> our organization is “qctrl” (<a href="https://github.com/qctrl" rel="nofollow noopener">https://github.com/qctrl</a>)</p>
<p>Affected repos currently:</p>
<p>qctrl/qctrl-deploy</p>
<p>qctrl/api2</p>
<p>There could be more but we haven’t had the time to search through our repos, can’t confirm if it’s every repo that has the issue</p>
<p>Thanks for looking into this</p>

**Comment by Yanjingzhu** [comment]
<p>Hi All, I have reported this issue in our internal channel, when there are any good news about it, I will update here at the first time. Sorry for any inconvenience. </p>

**Comment by AlekSi** [comment]
<p>I see those checks being skipped now instead of failing, so it was fixed?</p>

**Comment by cfarrend** [comment]
<p>Hey <a class="mention" href="/aleksi">@aleksi</a> same thing here, seems to show as “skipped” now. Might be a new feature?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Yeah, this issue has been fixed already. Skipped job should show skipped not failed now. Sorry for any inconvenience.  </p>
